src/main/java/org/broadinstitute/consent/http/service/MetricsService.java

```diff
@@ -160,4 +176,54 @@ public String getHeaderRow(Type type) {
     }
     return election;
   }
+
+  public DatasetMetrics generateDatasetMetrics(Integer datasetId) {
+
+    DatasetMetrics metrics = new DatasetMetrics();
+
+    //get datasetDTO with properties and data use restrictions
+    Set<DatasetDTO> datasets = dataSetDAO.findDatasetDTOWithPropertiesByDatasetId(datasetId);
+    if (datasets == null || datasets.isEmpty()) {
+       throw new NotFoundException("Dataset with specified ID does not exist.");
+    }
+
+    //find dars with the given datasetId in their list of datasetIds, datasetId is a String so it can be converted to jsonb in query
+    //convert all dars into smaller objects that only contain the information needed
+    List<DataAccessRequest> dars = dataAccessRequestDAO.findAllDataAccessRequestsForDatasetMetrics(Integer.toString(datasetId));
+    List<Object> darInfo = dars.stream().map(dar -> new Object() {
+      public Timestamp updateDate = dar.getUpdateDate();
+      public String projectTitle = dar.data.getProjectTitle();
+      public String darCode = dar.data.getDarCode();
+      public String nonTechRus = dar.data.getNonTechRus();
+      public String investigator = findPI(userPropertyDAO.findResearcherPropertiesByUser(dar.userId), dar.data.getResearcher()); 
+    }).collect(Collectors.toList());
+
+    //if there are associated dars, find associated access elections so we know how many and which dars are approved/denied
+    List<String> referenceIds = dars.stream().map(dar -> (dar.referenceId)).collect(Collectors.toList());
+    if (!referenceIds.isEmpty()) {
+      List<Election> elections = electionDAO.findLastElectionsByReferenceIdsAndType(referenceIds, "DataAccess");
+      metrics.setElections(elections);
+    } else {
+      metrics.setElections(Collections.emptyList());
+    }
+    metrics.setDataset(datasets.iterator().next());
+    metrics.setDars(darInfo);
+    return metrics;
+
+  }
+
+  public String findPI(List<UserProperty> props, String researcher) {
```

Comment:
See comment above. I think this would be more legible if it took a user id, queried for the user+props, and then with that information, figure out the PI name string to return.